x-pack/plugin/sql/sql-client/src/main/java/org/elasticsearch/xpack/sql/client/UriUtils.java

```diff
@@ -76,4 +76,30 @@ public static URI removeQuery(URI uri, String connectionString, URI defaultURI)
             throw new IllegalArgumentException("Invalid connection configuration [" + connectionString + "]: " + e.getMessage(), e);
         }
     }
+
+    public static URI appendSegmentToPath(URI uri, String segment) {
+        if (segment == null || segment.isEmpty() || "/".equals(segment)) {
+            return uri;
+        }
+        
+        String path = uri.getPath();
+        String concatenatedPath = "";
+        String cleanSegment = segment.startsWith("/") ? segment.substring(1) : segment;
+        
+        if (path == null || path.isEmpty()) {
+            path = "/";
+        }
+
+        if (path.charAt(path.length() - 1) == '/') {
+            concatenatedPath = path + cleanSegment;
+        } else {
+            concatenatedPath = path + "/" + cleanSegment;
+        }
+        try {
+            return new URI(uri.getScheme(), uri.getUserInfo(), uri.getHost(), uri.getPort(), concatenatedPath,
```

Comment:
Why not using new `URI(concatenatedPath)` ?

Reply:
Because `concatenatedPath` is only the `uri.getPath()` section of the initial URI. If one has `http://localhost:9200/es_rest/` path is `/es_rest/`. And this method's purpose is to add a new segment to the already existent path: `http://localhost:9200/es_rest/` + `/_sql` for example.